src/mongoserver.py

```diff
@@ -0,0 +1,131 @@
+# Copyright 2021 Canonical Ltd
+# See LICENSE file for licensing details.
+
+import secrets
+import string
+
+from tenacity import retry, stop_after_attempt, wait_exponential, RetryError
+from pymongo import MongoClient
+import logging
+
+
+logger = logging.getLogger(__name__)
+
+# We expect the mongodb container to use the
+# default ports
+MONGODB_PORT = 27017
+
+
+class MongoDB:
+    def __init__(self, config):
+        self.app_name = config["app_name"]
+        self.replica_set_name = config["replica_set_name"]
+        self.num_peers = config["num_peers"]
+        self.port = config["port"]
+        self.root_password = config["root_password"]
+        self.unit_ips = config["unit_ips"]
+
+    def client(self) -> MongoClient:
+        """Construct a client for the MongoDB database.
+
+        The timeout for all queries using this client object is 1 sec.
+
+        Returns:
+            A pymongo :class:`MongoClient` object.
+        """
+        return MongoClient(self.replica_set_uri(), serverSelectionTimeoutMS=1000)
+
+    @retry(stop=stop_after_attempt(5), wait=wait_exponential(multiplier=1, min=2, max=30))
+    def check_server_info(self, client: MongoClient):
+        """Repeatly checks to see if the server is ready, timing out after 20 tries
```

Comment:
Fix to 5 attempts according to the `retry` decorator's `stop` parameter.

Reply:
Oops my bad, I changed it to 5 for testing purposes, I'll change that back. Thanks Marcelo :)